src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnectorConfig.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.servicenow.connector;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.servicenow.util.ServiceNowConstants;
+import io.cdap.plugin.servicenow.util.Util;
+
+
+/**
+ * PluginConfig for ServiceNow Connector
+ */
+public class ServiceNowConnectorConfig extends PluginConfig {
+
+  @Name(ServiceNowConstants.PROPERTY_CLIENT_ID)
+  @Macro
+  @Description(" The Client ID for ServiceNow Instance.")
+  private final String clientId;
+
+  @Name(ServiceNowConstants.PROPERTY_CLIENT_SECRET)
+  @Macro
+  @Description("The Client Secret for ServiceNow Instance.")
+  private final String clientSecret;
+
+  @Name(ServiceNowConstants.PROPERTY_API_ENDPOINT)
+  @Macro
+  @Description("The REST API Endpoint for ServiceNow Instance. For example, https://instance.service-now.com")
+  private final String restApiEndpoint;
+
+  @Name(ServiceNowConstants.PROPERTY_USER)
+  @Macro
+  @Description("The user name for ServiceNow Instance.")
+  private final String user;
+
+  @Name(ServiceNowConstants.PROPERTY_PASSWORD)
+  @Macro
+  @Description("The password for ServiceNow Instance.")
+  private final String password;
+
+  public ServiceNowConnectorConfig(String clientId, String clientSecret, String restApiEndpoint,
+                                   String user, String password) {
+    this.clientId = clientId;
+    this.clientSecret = clientSecret;
+    this.restApiEndpoint = restApiEndpoint;
+    this.user = user;
+    this.password = password;
+  }
+
+  public String getClientId() {
+    return clientId;
+  }
+
+  public String getClientSecret() {
+    return clientSecret;
+  }
+
+  public String getRestApiEndpoint() {
+    return restApiEndpoint;
+  }
+
+  public String getUser() {
+    return user;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+
+  public void validateCredentials(FailureCollector collector) {
```

Comment:
similar method is also available in other config also. Reuse this one.

Reply:
It's better to use separate validation method here as the validation method which is defined in another config class includes the other validations as well which is not needed here.

Reply:
for similar validations you can call this method from that method and for rest of the validations keep that method as it is.

Reply:
Done